lib/logstash/inputs/tcp/decoder_impl.rb

```diff
@@ -53,10 +55,20 @@ def init_first_read(channel_addr, received)
       @address = extract_host_name(channel_addr) # name _or_ address of sender
       @port = channel_addr.get_port # outgoing port of sender (probably random)
     end
+    @sslsubject = extract_sslsubject(channel)
     @first_read = false
     filtered
   end
 
+  private
+  def extract_sslsubject(channel)
+    return nil unless @tcp.ssl_enable && @tcp.ssl_verify
+
+    channel.pipeline().get("ssl-handler").engine().getSession().getPeerPrincipal().getName()
+  rescue Exception => e
```

Comment:
let's not use the anti-pattern or rescuing everything
```suggestion
  rescue java.lang.Exception
```
also might be worth logging the exception at least on the debug level